core/src/main/scala/com/codacy/analysis/core/model/ResultsSet.scala

```diff
@@ -2,8 +2,16 @@ package com.codacy.analysis.core.model
 
 import java.nio.file.Path
 
-sealed trait ResultsSet
+import com.codacy.plugins.api.metrics.LineComplexity
 
-final case class ToolResults(tool: String, fileResults: Set[FileResults]) extends ResultsSet
+final case class ToolResults(tool: String, fileResults: Set[FileResults])
+final case class FileResults(filename: Path, results: Set[ToolResult])
 
-final case class FileResults(filename: Path, results: Set[ToolResult], metrics: Option[FileMetrics])
+final case class MetricsResult(results: Either[String, Set[FileWithMetrics]])
```

Comment:
On worker I ended up not having this class, since for the tool results we don't have this. I'm just using the Either.

Reply:
I'm sorry 😞 

here's a bear 🐻